layouts/shortcodes/ecosystem/registry/search-form.html

```diff
@@ -100,11 +111,174 @@
   <ul class="list-unstyled" id="search-results"></ul>
   <ul class="list-unstyled" id="default-body">
     {{ range $key, $value := $registry -}}
-      {{ $value = merge $value (dict "_key" $key) -}}
-      {{ partial "ecosystem/registry/entry" (dict "value" $value "languageNames" $languageNames) }}
+    {{ $value = merge $value (dict "_key" $key) -}}
+    {{ partial "ecosystem/registry/entry" (dict "value" $value "languageNames" $languageNames) }}
     {{ end -}}
   </ul>
+  <!-- Pagination Controls -->
+  <div id="pagination" class="text-center">
+    <button id="prev-page" class="btn btn-outline-secondary">Previous</button>
+    <div id="pagination-numbers" class="d-inline-block"></div>
+    <button id="next-page" class="btn btn-outline-secondary">Next</button>
+  </div>
+
+
+  <!-- total pages -->
+  <div id="pagination" class="text-center">
+    <div id="pagination-numbers" class="d-inline-block"></div>
+  </div>
+
+  <!-- Go to Page Section -->
+  <div class="text-center mt-3">
+    <input id="page-input" type="number" class="form-control d-inline-block w-auto" placeholder="Go to page" min="1"
+      max="${totalPages}">
+    <button class="btn btn-outline-primary" onclick="goToPage()">Go</button>
+    <p id="page-error" class="text-danger"></p>
+  </div>
+
+  <!-- Results per Page Selection -->
+  <div class="text-center mt-3">
+    <select id="results-per-page" class="form-select d-inline-block w-auto" onchange="changeResultsPerPage()">
+      <option value="10">10</option>
+      <option value="20" selected>20</option>
+      <option value="30">30</option>
+      <option value="50">50</option>
+    </select>
+  </div>
```

Comment:
Make 50 the default, also provide options 25, 50, 100, 200, to have much bigger spacing.

Add something that tells people what they are selecting (number of items per page)